### PR TITLE
fix(frontend): Replace license-checker with yarn licenses.

### PR DESCRIPTION
**Pull request by zijianjoy**
**Description of your changes:**

Context: https://github.com/kubeflow/pipelines/pull/8092

- Replace license-checker with yarn. We can detect dependency licenses using `yarn licenses` and `yarn licenses generate-disclaimer`.
- Use `media` instead of false for tailwindcss `darkMode`.
- Upgrade `caniuse-lite`
- [Upgrade @types/node and note-fetch in frontend server](https://github.com/kubeflow/pipelines/pull/8101/commits/52b1f2be2283f3e40a6c582cd02cac52e0227255)
- Add `npm ci` in Dockerfile for frontend server

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: frontend/scripts/yarn-licenses.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+npm install -D yarn
+
+# 2. Set up yarn: It will convert from package.json to yarn.lock
+npx yarn install
+
+# 3. List all packages with their license
+# npx yarn licenses list | less
+
+# # 4. Summarize licenses
+# yarn licenses list | grep License | sort -u | cut -d: -f2
```
Comment (zpChris) [inline frontend/scripts/yarn-licenses.sh:15]: Should these two lines be commented out or removed?

File: frontend/server/minio-helper.ts [hunk 2]
```diff
@@ -102,7 +102,7 @@ function extractFirstTarRecordAsStream() {
   const extract = tar.extract();
   const transformStream = new Transform({
     write: (chunk: any, encoding: string, callback: (error?: Error | null) => void) => {
```
Comment (zpChris) [inline frontend/server/minio-helper.ts:104]: nit: Replace unused parameter with `_`, as referenced in [this Stack Overflow post](https://stackoverflow.com/a/32198002) for example? (This is also a TypeScript suggestion on the VSCode UI.)
```suggestion
    write: (chunk: any, _, callback: (error?: Error | null) => void) => {
```

**Review by zpChris — COMMENTED** [review]
I left two small nits, after which I'll approve.

I'm assuming this issue also resolves the `403` warning described in #8092 (as you referenced in the description) - if there's a way I should double-check and test this on my end, please let me know. Thanks for fixing this!

**Comment by zpChris** [comment]
/lgtm

Thanks for fixing this issue! Let me know if I should `/approve` as well.

File: frontend/scripts/yarn-licenses.sh (hunk 1 above)
Comment (zijianjoy) [reply frontend/scripts/yarn-licenses.sh]: Sounds good, I moved them to the bottom for usage reference.

**Comment by zijianjoy** [comment]
Thank you @zpChris for the LGTM! I will merge this PR.

**Pull request merged**